src/charm.py

```diff
@@ -46,6 +47,13 @@ def __init__(self, *args: tuple[Any]) -> None:
         """Initialize the charm."""
         super().__init__(*args)
 
+        self._grafana_agent = COSAgentProvider(
```

Comment:
It's worth thinking should we start the snap service without this relation. We generally think the exporter service should be started only when it's required (i.e. there is consumers for the metric). Can you add some logic in the `configure` method to properly handle that?